qcodes/monitor/monitor.py

```diff
@@ -40,64 +41,69 @@ def _get_metadata(*parameters) -> Dict[str, Any]:
-            meta = deepcopy(_meta)
+        # Get the latest value from the parameter, respecting the max_val_age parameter
+        meta = {}
+        meta["value"] = str(parameter.get_latest())
+        if parameter.get_latest.get_timestamp() is not None:
+            meta["ts"] = parameter.get_latest.get_timestamp().timestamp()
         else:
-            raise ValueError("Input is not a parameter; Refusing to proceed")
-        # convert to string
-        meta['value'] = str(meta['value'])
-        if isinstance(meta["ts"], datetime.datetime):
-            meta["ts"] = time.mktime(meta["ts"].timetuple())
+            meta["ts"] = None
         meta["name"] = parameter.label or parameter.name
         meta["unit"] = parameter.unit
-        # find the base instrument in case this is a channel parameter
-        baseinst = parameter._instrument
-        while hasattr(baseinst, '_parent'):
-            baseinst = baseinst._parent
-        accumulator = metas.get(str(baseinst), [])
-        accumulator.append(meta)
-        metas[str(baseinst)] = accumulator
+
+        # find the base instrument that this parameter belongs to
+        baseinst = parameter.root_instrument
+        if baseinst is None:
+            metas["Unbound Parameter"].append(meta)
+        else:
+            metas[str(baseinst)].append(meta)
+
+    # Create list of parameters, grouped by instrument
     parameters_out = []
     for instrument in metas:
         temp = {"instrument": instrument, "parameters": metas[instrument]}
         parameters_out.append(temp)
+
     state = {"ts": ts, "parameters": parameters_out}
     return state
 
 
 def _handler(parameters, interval: int):
-
-    async def serverFunc(websocket, path):
+    """
+    Return the websockets server handler
+    """
+    async def server_func(websocket, path):
+        """
+        Create a websockets handler that sends parameter values to a listener
+        every "interval" seconds
+        """
         while True:
             try:
+                # Update the parameter values
                 try:
                     meta = _get_metadata(*parameters)
                 except ValueError as e:
                     log.exception(e)
                     break
-                log.debug(f"sending.. to {websocket}")
-                try:
-                    await websocket.send(json.dumps(meta))
-                # mute browser disconnects
-                except websockets.exceptions.ConnectionClosed as e:
-                    log.debug(e)
+                log.debug("sending.. to %r", websocket)
+                await websocket.send(json.dumps(meta))
+                # Wait for interval seconds and then send again
                 await asyncio.sleep(interval)
-            except CancelledError:
-                log.debug("Got CancelledError")
+            except (CancelledError, websockets.exceptions.ConnectionClosed):
+                log.debug("Got CancelledError or ConnectionClosed")
```

Comment:
this message on its own without the logging the actual exception is useless. Since the exception is caught and ignored by intention, lets at least log it with `log.exception(...)`

Reply:
I think this should just be log.debug("..." exc_info=True) so that it's still logged at debug level

Reply:
Agreed with both above. This exception is how the server signals that the browser was closed, or that a server shutdown was requested, so logging it as debug makes sense, but adding exc_info for extra info.